src/main/java/hexgui/gui/MessageDialogs.java

```diff
@@ -0,0 +1,390 @@
+    if (value == destructiveOption) result = 0;
+    else if (value == nonDestructiveOption) result = 1;
+    else result = 2;
+    return result;
+  }
+
+  public void showWarning(
+      Component parent, String mainMessage, String optionalMessage, boolean isCritical) {
+    showWarning(null, parent, mainMessage, optionalMessage, isCritical);
+  }
+
+  public void showWarning(
+      String disableKey,
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      boolean isCritical) {
+    if (checkDisabled(disableKey)) return;
+    int type;
+    if (isCritical) type = JOptionPane.WARNING_MESSAGE;
+    else type = JOptionPane.PLAIN_MESSAGE;
+    Object[] options = {"Close"};
+    Object defaultOption = options[0];
+    String title = "Warning" + " - " + m_applicationName;
+    show(
+        disableKey,
+        parent,
+        title,
+        mainMessage,
+        optionalMessage,
+        type,
+        JOptionPane.DEFAULT_OPTION,
+        options,
+        defaultOption,
+        -1);
+  }
+
+  public boolean showQuestion(
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      boolean isCritical) {
+    return showQuestion(null, parent, mainMessage, optionalMessage, destructiveOption, isCritical);
+  }
+
+  public boolean showQuestion(
+      String disableKey,
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String destructiveOption,
+      boolean isCritical) {
+    return showQuestion(
+        disableKey, parent, mainMessage, optionalMessage, destructiveOption, "Cancel", isCritical);
+  }
+
+  /**
+   * Show warning message to confirm destructive actions.
+   *
+   * @return true, if destructive was chosen; false if cancel was chosen.
+   */
+  public boolean showQuestion(
+      String disableKey,
+      Component parent,
+      String mainMessage,
+      String optionalMessage,
+      String affirmativeOption,
+      String cancelOption,
+      boolean isCritical) {
+    if (checkDisabled(disableKey)) return true;
+    Object[] options = new Object[2];
+    if (Platform.isMac()) {
+      options[0] = cancelOption;
+      options[1] = affirmativeOption;
+    } else {
+      options[0] = affirmativeOption;
+      options[1] = cancelOption;
+    }
+    Object defaultOption = affirmativeOption;
+    int type;
+    if (isCritical)
+      // No reason to show a warning icon for confirmation dialogs
+      // of frequent actions
+      type = JOptionPane.QUESTION_MESSAGE;
+    else type = JOptionPane.PLAIN_MESSAGE;
+    String title = "Question" + " - " + m_applicationName;
+    Object result =
+        show(
+            disableKey,
+            parent,
+            title,
+            mainMessage,
+            optionalMessage,
+            type,
+            JOptionPane.YES_NO_OPTION,
+            options,
+            defaultOption,
+            -1);
+    return (result == affirmativeOption);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.